src/editor/editor_main.cpp

```diff
@@ -18,22 +18,110 @@
 #include "editor/controller/editor_controller.hpp"
 
 #include "gettext.hpp"
+#include "gui/dialogs/editor/choose_addon.hpp"
+#include "gui/dialogs/message.hpp"
 #include "filesystem.hpp"
 #include "editor/action/action_base.hpp"
-
-#include <boost/algorithm/string/replace.hpp>
+#include "serialization/parser.hpp"
+#include "serialization/preprocessor.hpp"
 
 lg::log_domain log_editor("editor");
 
 namespace editor {
 
+void initialize_addon(const std::string& addon_id)
+{
+	std::string addon_dir = filesystem::get_addons_dir() + "/" + addon_id;
+
+	// create folders
+	filesystem::create_directory_if_missing(addon_dir);
+	filesystem::create_directory_if_missing(addon_dir + "/maps");
+	filesystem::create_directory_if_missing(addon_dir + "/scenarios");
+	filesystem::create_directory_if_missing(addon_dir + "/images");
+	filesystem::create_directory_if_missing(addon_dir + "/images/units");
+	filesystem::create_directory_if_missing(addon_dir + "/masks");
+	filesystem::create_directory_if_missing(addon_dir + "/music");
+	filesystem::create_directory_if_missing(addon_dir + "/sounds");
+	filesystem::create_directory_if_missing(addon_dir + "/translations");
+	filesystem::create_directory_if_missing(addon_dir + "/units");
+	filesystem::create_directory_if_missing(addon_dir + "/utils");
+
+	// create files
+	// achievements
+	{
+		filesystem::scoped_ostream stream = filesystem::ostream_file(addon_dir + "/achievements.cfg");
+		*stream << "";
+	}
+
+	// _server.pbl
+	{
+		filesystem::scoped_ostream stream = filesystem::ostream_file(addon_dir + "/_server.pbl");
+		*stream << "";
+	}
+
+	// a basic _main.cfg
+	{
+		filesystem::scoped_ostream stream = filesystem::ostream_file(addon_dir + "/_main.cfg");
+		*stream << "#textdomain wesnoth-" << addon_id << "\n"
+				<< "[textdomain]" << "\n"
+				<< "\tname=\"wesnoth-" << addon_id << "\"\n"
```

Comment:
The rest of the file is indented with tabs though.

Reply:
Why would this one file be different from all the others? All the other files are indented with tabs.

Reply:
All other files are indented with spaces... I guess it's not clear I'm talking about the file being written here. I'm not suggesting to randomly change the indentation of the source code.

Reply:
In case it's still not clear what I'm talking about. I'm talking about the content of the _main.cfg file that is being written here. It should be written so that wmlindent agrees with the indentation.

Reply:
Updated.

Reply:
I assume this _main.cfg is written once and never again touched?

The `#ifdef MULTIPLAYER` is going to be wrong for about 50% of add-ons. I'm also wondering if the `[textdomain]` tag is positioned correctly. Does it belong inside the `#ifdef` along with the `[binary_path]`? I'm not entirely sure.

Reply:
It's written once when a new add-on is created, and after that it's only ever changed when the add-on ID is changed.

Reply:
Is it completely overwritten or are changes kept? I was thinking it can be adjusted if it doesn't fit the addon.

Reply:
> after that it's only ever changed when the add-on ID is changed.

How does this work? Just read as text and find-and-replace? (What exactly depends on the add-on's ID, anyway?)

Reply:
It edits the existing file to replace all instances of of `/<old_addon_id>` with `/<new_addon_id>`.

Reply:
So in other words it is indeed just read as text and find-and-replace.

Reply:
How would I determine that?

Reply:
I'm not exactly sure… maybe try putting it inside the define and see if the translations still work?

Reply:
Is there a reason to think that the documented example is wrong? My guess would be that having conditionally defined textdomains would be problematic, but I've never setup translations before so that's not something I can easily go and do.

Reply:
Mostly it's that `[textdomain]` and `[binary_path]` feel like similar constructs to me, so it's a bit weird that they have different recommendations.

Reply:
I could also just remove it from being added here - I assume anyone who's going through the process of adding translations can add it themselves easily enough later.

Reply:
Given the existing documentation/example as well as Ravana's response on Discord:

> it probably isnt needed there, but with addon separation logic there should be no harm either

I'm going to leave it as it is, unless someone knows of a more definitive reason that it's a problem.

Reply:
To be clear, I don't think either way is a problem. I just think this is a good opportunity to show an example of "best practice", and at the same time re-evaluate whether the current way really is "best practice".